CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+# platform(s), but allow optionally enabling.
+# For reference, note that the default varies by platform, e.g.
+# * CentOS 7.9.2009's GCC 6.3, `_GLIBCXX_USE_CXX11_ABI` is
+#   always `0` and cannot be overridden.
+# * Ubuntu 18.04's GCC 6.5, `_GLIBCXX_USE_CXX11_ABI` defaults
+#   to `1`, but can be overridden.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+    option(OPENASSETIO_GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" OFF)
+endif ()
+
+# Whether to use RPATH or RUNPATH runtime search path behavior.
+if (IS_GCC_OR_CLANG)
+    option(OPENASSETIO_ENABLE_NEW_DTAGS "If enabled, RUNPATH will be set, overriding RPATH" OFF)
+endif ()
+
+# Testing instrumentation.
+if (IS_GCC_OR_CLANG)
+    # Coverage reporting (for tests).
+    option(OPENASSETIO_ENABLE_COVERAGE "Enable coverage reporting for gcc/clang" OFF)
+
+    # Sanitizers (primarily for tests)
+    option(OPENASSETIO_ENABLE_SANITIZER_ADDRESS "Enable address sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_LEAK "Enable leak sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+        "Enable undefined behavior sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_THREAD "Enable thread sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_MEMORY "Enable memory sanitizer" OFF)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Enable ctest testing
+
+# To correctly generate the DartConfiguration.tcl, include the CTest
+# CMake file. By doing so 'ctest -F memcheck' works.
+# TODO(DF): ...but also creates many many pointless targets.
+#include(CTest)  # Automatically calls enable_testing()
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Provides install directory variables as defined by the GNU Coding
+# Standards.
+
+include(GNUInstallDirs)
+
+#-----------------------------------------------------------------------
+# Generate OpenAssetIOConfig.cmake et al. for bundling as a CMake lib.
+
+# Targets exported to ${PROJECT_NAME}_EXPORTED_TARGETS will be available
+# for import using find_package(OpenAssetIO).
+
+include(CMakePackageConfigHelpers)
+
+set(version_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+# Version fetched from the top level project()
+write_basic_package_version_file(
+    ${version_config_file}
+    # Version range supported only if generated by CMake >= 3.19.2.
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    cmake/Config.cmake.in
+    ${project_config_file}
+    INSTALL_DESTINATION ${config_install_dir}
+    # TODO(DF): PATH_VARS?
+)
+
+install(
+    EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS
+    DESTINATION ${config_install_dir}
```

Comment:
The value of `config_install_dir` is a relative path on my box, specifically `lib/cmake/OpenAssetIO`

The contents of `OpenAssetIOConfig.cmake`, once installed, is
```
# SPDX-License-Identifier: Apache-2.0
# Copyright 2013-2022 The Foundry Visionmongers Ltd

set(OpenAssetIO_VERSION 0.0.0)


####### Expanded from @PACKAGE_INIT@ by configure_package_config_file() #######
####### Any changes to this file will be overwritten by the next CMake run ####
####### The input file was Config.cmake.in                            ########

get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)

macro(set_and_check _var _file)
  set(${_var} "${_file}")
  if(NOT EXISTS "${_file}")
    message(FATAL_ERROR "File or directory ${_file} referenced by variable ${_var} does not exist !")
  endif()
endmacro()

macro(check_required_components _NAME)
  foreach(comp ${${_NAME}_FIND_COMPONENTS})
    if(NOT ${_NAME}_${comp}_FOUND)
      if(${_NAME}_FIND_REQUIRED_${comp})
        set(${_NAME}_FOUND FALSE)
      endif()
    endif()
  endforeach()
endmacro()

####################################################################################

# TODO(DF): add here any required find_dependency(), E.g.
# include(CMakeFindDependencyMacro)
# find_dependency(Imath  HINTS )

set_and_check (OpenAssetIO_DIR "${PACKAGE_PREFIX_DIR}")
set_and_check (OpenAssetIO_INCLUDE_DIR "${PACKAGE_PREFIX_DIR}/include")
set_and_check (OpenAssetIO_LIBRARY_DIR "${PACKAGE_PREFIX_DIR}/lib")

include ("${CMAKE_CURRENT_LIST_DIR}/OpenAssetIOTargets.cmake")

check_required_components ("OpenAssetIO")
```

Reply:
Haha... in my foggy cold head I incorrectly read all these variables as being grounded in the PROJECT_BINARY_DIR
```
set(version_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
set(project_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
```

whereas the `config_install_dir` is not. Hence relative.

That config file looks fine to me. I should've said the non-configuration specific target config, but since there is the get_filename_component working out the prefix, then I'm sure it'll be fine.

Nothing to see here. Sorry for the red herring.